src/source-updater.js

```diff
@@ -164,4 +177,18 @@ export default class SourceUpdater {
       this.sourceBuffer_.abort();
     }
   }
+
+  handleBufferMaxed_(event) {
+    videojs.log.warn('SourceBuffer exceeded quota; attempting to recover');
+    this.deferredCallback_ = this.pendingCallback_;
```

Comment:
Naive alternative suggestion(I still need to think this through again once I get to the end of the PR):

Couldn't you pass in the pendingCallback alone(rather than done)? Then call pendingCallback after resolving the failed append? `onUpdateendCallback_` would then resolve anything else in the queue?

Reply:
Or another alternative would be to add a flag to `onUpdateendCallback_` that would fail fast(not actually run the rest of the callbacks) while we are resolving the failed append. Then once we successfully resolve that append, we can turn the flag off and allow the `onUpdateendCallback_` to continue to run normally. This might be a bit more complicated in practice though, since we're not 100% sure our next attempt to append after trimming the back buffer will work.

Reply:
IIRC simply removing `|| this.pendingCallback_` from [here](https://github.com/squarebracket/videojs-contrib-hls/blob/79f53fed214a241df7e02801fb52deb1ca662ce1/src/source-updater.js#L131) caused problems, but I'm open to trying it again.

Reply:
(I said this because your solutions seem to imply that requirement)

Reply:
Copying over gesinger's suggestion from Slack:

> Since pendingCallback_ should only be referenced internal to this class, it may make sense to have an additional property like waitingToAppend, and the code paths which normally rely on pendingCallback_ for determining whether something is updating/etc. also consider if we’re waitingToAppend or not before basing the status on pendingCallback_ and/or the updateend.

Which, if we follow, would involve adding `waitingToAppend` to that method.